### DISCUSSION TITLE
cd into directory prior to Docker step

### DISCUSSION THREAD

**Discussion by Happy-Ferret** [community/community#25877] [Actions]
<p>Coming from GitLab CI/CD, I’m used to having a “before_script” step that CDs into my project’s “WebApplication1Core” folder (the Dockerfile is simply set up that way and I’d prefer not changing it/was having issues when I changed it).</p>
<p>How can this be achieved in GitHub Actions?</p>
<p>Simply adding a “run” step that did this didn’t work. The progress wasn’t retained and the Docker step was still run from the repository’s root directory.</p>
<pre><code>steps:
    - uses: actions/checkout@v1
    - name: Sign into Docker registry
      uses: actions/docker/login@master
      env:
        DOCKER_USERNAME: ${{ secrets.DOCKER_USER }}
        DOCKER_PASSWORD: ${{ secrets.DOCKER_PW }}
        DOCKER_REGISTRY_URL: docker.pkg.github.com
    - name: Access Source directory
      run: cd WebApplication1Core
    - name: Ensure we are in the correct directory context
      run: pwd
    - name: Build Docker image
      uses: actions/docker/cli@master
      with:
        args: 'build -t docker.pkg.github.com/happy-ferret/ci-test-mr/webapplication1core:latest -f WebApplication1Core/Dockerfile .'
    - name: Push Docker image
      uses: actions/docker/cli@master
      env:
        DOCKER_USERNAME: ${{ secrets.DOCKER_USER }}
        DOCKER_PASSWORD: ${{ secrets.DOCKER_PW }}
        DOCKER_REGISTRY_URL: docker.pkg.github.com
      with:
        args: 'push docker.pkg.github.com/happy-ferret/ci-test-mr/webapplication1core:latest'
</code></pre>
<p>By the time I run into the “pwd” step, the directory is already reset to the root folder. The Docker build step fails because it cannot find the Dockerfile. If I add the full path to the Docker file instead, something else breaks further down the line.</p>

**Comment by oldskool** [comment, marked as answer]
<p>Please note that the <a href="https://github.com/actions/docker#important" rel="nofollow noopener">actions/docker</a> runner has been deprecated in favor of the run command. Since <a href="https://help.github.com/en/articles/software-in-virtual-environments-for-github-actions" rel="nofollow noopener">Docker is pre-installed in the virtual environment</a>, you should be able to simply do something like:</p>
<pre><code>steps:
- name: Build Docker image
    run: |
      cd WebApplication1Core
      docker build -t docker.pkg.github.com/happy-ferret/ci-test-mr/webapplication1core:latest .
</code></pre>

**Comment by Happy-Ferret** [comment]
<p>That did indeed do the trick. Thank you!</p>
